File: internal/flowdoc/testdata/verify-symbols/sample.c
/* C fixture for verify symbol-grammar tests. */

#include <stdio.h>

struct Frame {
    int id;
    unsigned char data[8];
};

typedef struct Frame frame_t;

#define MAX_FRAMES 128

static int helper_count(int n) {
    return n + 1;
}

int run_pipeline(int argc, char **argv) {
    /* call site that should NOT be matched as a declaration */
    return helper_count(argc);
}

void emit_frame(struct Frame *f) {
    if (f) printf("%d\n", f->id);
}

File: internal/flowdoc/testdata/verify-symbols/sample.rs
// Rust fixture for verify symbol-grammar tests.

pub struct CanMonitor {
    name: String,
}

pub enum FrameKind {
    Data,
    Remote,
}

pub const MAX_FRAMES: usize = 128;

impl CanMonitor {
    pub fn new(name: String) -> Self {
        Self { name }
    }
}

pub fn run_single_frame_mode(cfg: &str) -> Result<(), String> {
    Ok(())
}

pub async fn run_replay_mode(file: &str) -> Result<(), String> {
    Ok(())
}

fn load_config() -> Result<String, String> {
    Ok(String::new())
}

macro_rules! trace_call {
    ($x:expr) => { println!("{}", $x) };
}
